### DISCUSSION TITLE
Hetzner cloud userdata

### DISCUSSION THREAD

**Discussion by Kenzim** [siderolabs/talos#4944] [Q&A] [answered]
Hi, 

I followed the tutorial on how to make a cluster on Hetzner cloud - I had the Talos iso uploaded to my project, and then booted the control plane servers and workers using that. I loaded the configuration YAML files into the user data for the servers like the tutorial shows, but the servers didn't seem to pick it up. I had to manually apply the configuration to the servers, but this as far as I am aware is temporary and the config is stored in memory. Is there anything I have to do to make it read the config from Hetzner's user data provider? Any help would be appreciated.

Thanks

**Comment by sergelogvinov** [comment, marked as answer]
Hello, ISO(CD) version uses metal platform by default. You can switch it in grub menu at boot time

```
talos.platform=hcloud
```

Second way is create a VM and make snapshot. And use this snapshot to bootstrap new VMs.
https://www.talos.dev/docs/v0.14/cloud-platforms/hetzner/#rescue-mode 

Third - use the hashicorp packer https://www.talos.dev/docs/v0.14/cloud-platforms/hetzner/#packer

**Reply by Kenzim** [reply]
Thanks for the quick answer. So I could change the platform in the ISO and then use that instead?

Thanks

**Reply by sergelogvinov** [reply]
Easiest way is snapshot.

ISO: You need to rebuild iso image, and change the /grub/grub.cfg
It looks like:
```conf
menuentry "Talos ISO" {
	set gfxmode=auto
	set gfxpayload=text
	linux /boot/vmlinuz init_on_alloc=1 slab_nomerge pti=on panic=0 consoleblank=0 printk.devkmsg=on earlyprintk=ttyS0 console=tty0 console=ttyS0 talos.platform=metal
	initrd /boot/initramfs.xz
```

you need to change ```talos.platform=metal``` -> ```talos.platform=hcloud ```

At boot time, you can put ```e``` key and edit the menu too.
